src/ebpf.h

```diff
@@ -217,4 +217,56 @@ struct ebpf_inst
 #define EBPF_OP_ATOMIC32_STORE (EBPF_CLS_STX | EBPF_MODE_ATOMIC | EBPF_SIZE_W)
 #define EBPF_OP_ATOMIC_STORE (EBPF_CLS_STX | EBPF_MODE_ATOMIC | EBPF_SIZE_DW)
 
+typedef enum bpf_map_type
+{
+    BPF_MAP_TYPE_UNSPEC = 0, ///< Unspecified map type.
+    BPF_MAP_TYPE_HASH = 1,   ///< Hash table.
+    BPF_MAP_TYPE_ARRAY = 2,  ///< Array, where the map key is the array index.
+    BPF_MAP_TYPE_PROG_ARRAY =
+        3, ///< Array of program fds usable with bpf_tail_call, where the map key is the array index.
+    BPF_MAP_TYPE_PERCPU_HASH = 4,      ///< Per-CPU hash table.
+    BPF_MAP_TYPE_PERCPU_ARRAY = 5,     ///< Per-CPU array.
+    BPF_MAP_TYPE_HASH_OF_MAPS = 6,     ///< Hash table, where the map value is another map.
+    BPF_MAP_TYPE_ARRAY_OF_MAPS = 7,    ///< Array, where the map value is another map.
+    BPF_MAP_TYPE_LRU_HASH = 8,         ///< Least-recently-used hash table.
+    BPF_MAP_TYPE_LPM_TRIE = 9,         ///< Longest prefix match trie.
+    BPF_MAP_TYPE_QUEUE = 10,           ///< Queue.
+    BPF_MAP_TYPE_LRU_PERCPU_HASH = 11, ///< Per-CPU least-recently-used hash table.
+    BPF_MAP_TYPE_STACK = 12,           ///< Stack.
+    BPF_MAP_TYPE_RINGBUF = 13          ///< Ring buffer.
```

Comment:
These numbers are Windows numbers and are different on Linux.  Is this what you intend?